naxsi_src/naxsi_config.c

```diff
@@ -114,15 +104,15 @@ naxsi_score(ngx_conf_t* r, ngx_str_t* tmp, ngx_http_rule_t* rule)
       NX_LOG_DEBUG(
         _debug_score, NGX_LOG_EMERG, r, 0, "XX-(debug) special scoring rule (%s)", tmp_ptr);
 
-      return_value_if(!(tmp_end = strchr(tmp_ptr, ':')), NGX_CONF_ERROR);
+      return_value_if(NULL == (tmp_end = strchr(tmp_ptr, ':')), NGX_CONF_ERROR);
 
       return_value_if((len = tmp_end - tmp_ptr) < 1, NGX_CONF_ERROR);
 
-      return_value_if(!(sc = ngx_array_push(rule->sscores)), NGX_CONF_ERROR);
+      return_value_if(NULL == (sc = ngx_array_push(rule->sscores)), NGX_CONF_ERROR);
 
-      return_value_if(!(sc->sc_tag = ngx_pcalloc(r->pool, sizeof(ngx_str_t))), NGX_CONF_ERROR);
+      return_value_if(NULL == (sc->sc_tag = ngx_pcalloc(r->pool, sizeof(ngx_str_t))), NGX_CONF_ERROR);
 
-      return_value_if(!(sc->sc_tag->data = ngx_pcalloc(r->pool, len + 1)), NGX_CONF_ERROR);
+      return_value_if(NULL == (sc->sc_tag->data = ngx_pcalloc(r->pool, len + 1)), NGX_CONF_ERROR);
```

Comment:
why this is an issue?

Reply:
That is [c4706](https://learn.microsoft.com/en-us/cpp/error-messages/compiler-warnings/compiler-warning-level-4-c4706?view=msvc-170), the fix is from this part of description:

> If you intend to make your test value the result of an assignment, test to ensure that the assignment is non-zero or not null.

Reply:
oh, that is some bs warning.